lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -135,79 +135,98 @@ static bool HasReflectionInfo(ObjectFile *obj_file) {
   return hasReflectionSection;
 }
 
+SwiftLanguageRuntime::NativeReflectionContext *
+SwiftLanguageRuntime::GetReflectionContext() {
+  if (!m_initialized_reflection_ctx)
+    SetupReflection();
+  return m_reflection_ctx.get();
+}
+
 void SwiftLanguageRuntime::SetupReflection() {
-  reflection_ctx.reset(new NativeReflectionContext(this->GetMemoryReader()));
+  std::lock_guard<std::recursive_mutex> lock(m_add_module_mutex);
+
+  m_reflection_ctx.reset(new NativeReflectionContext(this->GetMemoryReader()));
+  m_initialized_reflection_ctx = true;
 
   auto &target = m_process->GetTarget();
   auto exe_module = target.GetExecutableModule();
-  if (!exe_module)
-    return;
-  auto *obj_file = exe_module->GetObjectFile();
-  if (!obj_file)
-      return;
-  if (obj_file->GetPluginName().GetStringRef().equals("elf"))
-    return;
-  Address start_address = obj_file->GetBaseAddress();
-  auto load_ptr = static_cast<uintptr_t>(start_address.GetLoadAddress(&target));
-
-  // Bail out if we can't read the executable instead of crashing.
-  if (load_ptr == 0 || load_ptr == LLDB_INVALID_ADDRESS)
+  if (!AddModuleToReflectionContext(exe_module)) {
+    m_reflection_ctx.reset();
```

Comment:
`m_initialized_reflection_ctx = false` ? 

Or should you just get rid of this other variable and just check the unique_pointer?

Reply:
> m_initialized_reflection_ctx = false ?

No, `m_initialized_reflection_ctx` signals that `SetupReflection()` has already been run. If it failed here (while loading the main executable module) we shouldn't try reading it over and over again.

> Or should you just get rid of this other variable and just check the unique_pointer?

On ELF, reflection is generally disabled, so reflection_context will be `nullptr`, but again, we don't want to re-check on every request.